go.mod

```diff
@@ -21,8 +21,8 @@ require (
 	github.com/hashicorp/go-retryablehttp v0.7.2
 	github.com/hashicorp/golang-lru v0.5.4
 	github.com/jongio/azidext/go/azidext v0.4.0
-	github.com/onsi/ginkgo/v2 v2.8.4
-	github.com/onsi/gomega v1.27.1
+	github.com/onsi/ginkgo/v2 v2.9.1
```

Comment:
We would also need to update https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/c0dab22666648ea50e91cd9b73fa0b7113826cd1/Makefile#L108